fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/impl/datagen/FabricModel.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.datagen;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import net.minecraft.data.client.Model;
+import net.minecraft.data.client.TextureKey;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.datagen.v1.model.builder.ItemModelBuilder;
+import net.fabricmc.fabric.api.datagen.v1.model.property.DisplayBuilder;
+import net.fabricmc.fabric.api.datagen.v1.model.property.ElementBuilder;
+import net.fabricmc.fabric.api.datagen.v1.model.property.OverrideBuilder;
+
+public interface FabricModel {
+	default Model withDisplay(DisplayBuilder.Position position, DisplayBuilder builder) {
```

Comment:
All methods on impl interfaces that are mixined onto a class should be prefixed with `fabric_`